Framework/Algorithms/src/NormaliseToMonitor.cpp

```diff
@@ -561,99 +662,123 @@ void NormaliseToMonitor::normaliseBinByBin(
   auto outputEvent =
       boost::dynamic_pointer_cast<EventWorkspace>(outputWorkspace);
 
-  // Get hold of the monitor spectrum
-  const auto &monX = m_monitor->binEdges(0);
-  auto monY = m_monitor->counts(0);
-  auto monE = m_monitor->countStandardDeviations(0);
-  // Calculate the overall normalization just the once if bins are all matching
-  if (m_commonBins)
-    this->normalisationFactor(monX, monY, monE);
-
-  const size_t numHists = inputWorkspace->getNumberHistograms();
-  auto specLength = inputWorkspace->blocksize();
-  // Flag set when a division by 0 is found
-  bool hasZeroDivision = false;
-  Progress prog(this, 0.0, 1.0, numHists);
-  // Loop over spectra
-  PARALLEL_FOR_IF(
-      Kernel::threadSafe(*inputWorkspace, *outputWorkspace, *m_monitor))
-  for (int64_t i = 0; i < int64_t(numHists); ++i) {
-    PARALLEL_START_INTERUPT_REGION
-    prog.report();
-
-    const auto &X = inputWorkspace->binEdges(i);
-    // If not rebinning, just point to our monitor spectra, otherwise create new
-    // vectors
-
-    auto Y = (m_commonBins ? monY : Counts(specLength));
-    auto E = (m_commonBins ? monE : CountStandardDeviations(specLength));
-
-    if (!m_commonBins) {
-      // ConvertUnits can give X vectors of all zeros - skip these, they cause
-      // problems
-      if (X.back() == 0.0 && X.front() == 0.0)
-        continue;
-      // Rebin the monitor spectrum to match the binning of the current data
-      // spectrum
-      VectorHelper::rebinHistogram(
-          monX.rawData(), monY.mutableRawData(), monE.mutableRawData(),
-          X.rawData(), Y.mutableRawData(), E.mutableRawData(), false);
-      // Recalculate the overall normalization factor
-      this->normalisationFactor(X, Y, E);
-    }
+  std::shared_ptr<SpectrumInfo> inputSpecInfo;
+  std::shared_ptr<SpectrumInfo> monitorSpecInfo;
+  if (m_syncScanInput) {
+    inputSpecInfo =
+        std::make_shared<SpectrumInfo>(inputWorkspace->spectrumInfo());
+    monitorSpecInfo = std::make_shared<SpectrumInfo>(m_monitor->spectrumInfo());
```

Comment:
This takes copies of `SpectrumInfo`? Is this needed? Why does this need to be in an `if`? Getting the reference is cheap I think, even if it is not needed later.

Reply:
I've removed this now is is not needed. I was thinking an error was thrown if no instrument is present, but getting the reference is not an issue.